Appsilon/shiny.benchmark
# Adding ability to run tests multiple times

### Link to the Issue

close #14 

### Definition of Done

User is allowed to run the tests more than once using the `n_rep` argument.

### How to test changes

Steps for testing:

- unzip file
- open app project under app
- from terminal run `bash ../setting_branches.sh` to create all the branches we are interested
- you may need to install shiny.performance manually (build source package and install it)
- run `bash ../run_tests.sh` to run all tests (the last 2 are the important ones)

You should see an info about which iteration we are running. Also, the output should be a list of data.frames for each commit in the list.

[all_tests.zip](https://github.com/Appsilon/experimental.performance/files/9852639/all_tests.zip)

### Tasks for PR author


### General policy:

- If applicable - add instructions for testing
- If there’s no issue, create it. Each issue needs to be well defined and described.
- All interaction with a user, user-facing messages, plots, reports etc. are written from the perspective of the person using or receiving it. They are understandable and helpful to this person. If a user sees an error message, there is a call to action, i.e. the user knows what to do to fix it.
- README, other documentation and code comments that we have is updated with all information related to the change.
- All code has been peer-reviewed before merging into any main branch
- All changes have been merged into the main branch we use for development.
- Continuous integration checks (linter, unit tests, integration tests) are configured and pass.
- Optional: unit tests added for all new or changed logic.
- All task requirements satisfied. If not describe it here. The reviewer is responsible to verify each aspect of the task.
- Change covers only things in task. Please create new PR if you want to fix something else.
